src/MSALWrapper/AuthFlow/Broker.cs

```diff
@@ -73,65 +70,46 @@ private enum GetAncestorFlags
-                    account,
-                    this.pcaWrapper,
-                    this.errors);
-
-                if (tokenResult != null)
-                {
-                    return (tokenResult, this.errors);
-                }
+            TokenResult tokenResult = await CachedAuth.GetTokenAsync(
+                this.logger,
+                this.scopes,
+                account,
+                this.pcaWrapper,
+                this.errors);
 
-                try
-                {
-                    tokenResult = await TaskExecutor.CompleteWithin(
-                        this.logger,
-                        this.interactiveAuthTimeout,
-                        $"{this.Name()} interactive auth",
-                        this.GetTokenInteractive(account),
-                        this.errors).ConfigureAwait(false);
-                }
-                catch (MsalUiRequiredException ex)
-                {
-                    this.errors.Add(ex);
-                    this.logger.LogDebug($"Initial {this.Name()} auth failed. Trying again with claims from exception.\n{ex.Message}");
-
-                    tokenResult = await TaskExecutor.CompleteWithin(
-                        this.logger,
-                        this.interactiveAuthTimeout,
-                        $"{this.Name()} interactive auth (with extra claims)",
-                        this.GetTokenInteractiveWithClaims(ex.Claims),
-                        this.errors).ConfigureAwait(false);
-                }
-            }
-            catch (MsalServiceException ex)
+            if (tokenResult != null)
             {
-                this.logger.LogWarning($"MSAL Service Exception! (Not expected)\n{ex.Message}");
-                this.errors.Add(ex);
+                return tokenResult;
             }
-            catch (MsalClientException ex)
+
+            try
             {
-                this.logger.LogWarning($"Msal Client Exception! (Not expected)\n{ex.Message}");
-                this.errors.Add(ex);
+                tokenResult = await TaskExecutor.CompleteWithin(
+                    this.logger,
+                    this.interactiveAuthTimeout,
+                    $"{this.Name()} interactive auth",
+                    this.GetTokenInteractive(account),
+                    this.errors).ConfigureAwait(false);
             }
-            catch (NullReferenceException ex)
+            catch (MsalUiRequiredException ex)
             {
-                this.logger.LogWarning($"Msal unexpected null reference! (Not Expected)\n{ex.Message}");
                 this.errors.Add(ex);
+                this.logger.LogDebug($"initial {this.Name()} auth failed. Trying again with claims from exception.\n{ex.Message}");
+
+                tokenResult = await TaskExecutor.CompleteWithin(
+                    this.logger,
+                    this.interactiveAuthTimeout,
+                    $"{this.Name()} interactive auth (with extra claims)",

```

Comment:
Now we have a base class. Is it a good time to change it to a property `this.Name`?

Reply:
Abstract properties do look to be possible! we can make that change in a follow up PR! That would simplify each authflow a tad.